app/Helpers/ReceivesReplies.php

```diff
@@ -25,7 +25,11 @@ public function latestReplies(int $amount = 5)
 
     public function deleteReplies()
     {
-        $this->repliesRelation()->delete();
+        // We need to explicitly iterate over the repies and delete them
+        // separately because all related models need to be deleted.
+        foreach ($this->repliesRelation()->get() as $reply) {
+            $reply->delete();
+        }
```

Comment:
Bit unfortunate that we can't use foreign keys for this. Beginning to dislike polymorphic relationships 😄